### ISSUE TITLE
css transitions none,  prevents showing next months

### ISSUE THREAD

**Issue by sag1v** [airbnb/react-dates#976]
Hi, I've encountered a nasty bug with one of our apps which uses `react-dates`.  
tl;dr:  
If you will disable CSS `transitions` i.e:  
```
* {
    -webkit-transition: none !important;
    transition: none !important;
    -webkit-animation: none !important;
    animation: none !important;
}
```

This will prevent the months from being populated, you will get stuck showing only the first couple of months that are initially rendered (4 months overall i think).  

The reason for this is that [`updateStateAfterMonthTransition` ](https://github.com/airbnb/react-dates/blob/master/src/components/DayPicker.jsx#L507) won't calculate the next months.  

I understand the logic behind this, but i don't think i fully agree with this approach.  
Why should a `CSS` rule should break the behavior of the component.  
If there are no transitions available on the DOM, then we might just render the next months without the nice slide effect.
In our case, this `CSS` rule came from our customers which our app runs on their website.
what are your thoughts about that? :) 

[Here](https://codesandbox.io/s/5vpk9k37j4) is an example of this behavior

**Comment by majapw** [comment]
Hi @sag1v, if you pass in `transitionDuration={0}` into the datepicker, it should shortcircuit that logic and fix the issue.

Um, as for
> Why should a CSS rule should break the behavior of the component.
The whole modular react thing we follow is an encapsulation of behavior and an attempt to prevent side-effecty behavior. Global CSS rules are tricky in this way because it's not ... all that surprising that they have unexpected consequences. There's no way for the component to know about css defined everywhere else in the app (and update accordingly) but it can respond to explicit props... so I'd recommend using the `transitionDuration` prop for this scenario.

Is that an adequate solution?

**Comment by sag1v** [comment]
> if you pass in `transitionDuration={0}` into the datepicker, it should shortcircuit that logic and fix the issue.

@majapw Thanks, but if you mean adding the `transitionDuration={0}` to the `DateRangePicker`, i'm afraid it's not fixing the issue (you can try it on the snippet i posted above).

As for your explanation on  

> Why should a CSS rule should break the behavior of the component.

I fully agree with your statement, though i still think that some `JavaScript` logic of the component like displaying the next month when clicked on the next button should't depend on `CSS` rules.

I mean, `CSS` behavior will obviously depend on other / global `CSS`  rules.  
But adding a new date object to a JavaScript  array, should defiantly not be impact by a `CSS` rule.

> css defined everywhere else in the app 

I understand that, and i agree. 
On top of my head, one quick (and maybe dirty) solution could be attacking it from a css angle.  
Maybe add an explicit css rule of the transition we need (that will override global css).  

Anyway this is an edge case, but when it happens to you it's hard to debug :smiley_cat:

**Comment by YvanGuidoin** [comment]
Problem maybe linked but same solution transitionDuration={0} correct the problem:

> Warning: Failed prop type: Invalid prop `modifiers` of type `Object` supplied to `CalendarDay`, expected instance of `Set`.

It happens only AFTER the month change on re rendering, the `CalendarDay.render()` is called again with wrong props, but if `transitionDuration={0}` it never happens,

In case in changes something, I have a custom `isOutsideRange` logic on my SingleDayPicker
